### PR TITLE
add __next__ method to RawTextIterableDataset

### PR DESCRIPTION
**Pull request by zhangguanheng66**

### CODE REVIEW

File: torchtext/experimental/datasets/raw/common.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def __iter__(self):
                 break
             yield item
 
+    def __next__(self):
```
Comment (cpuhrsch) [inline torchtext/experimental/datasets/raw/common.py:47]: One concern I have with this approach is the following:

What happens if you do this:
```
for line in dataset:
    print(line)
    print(next(dataset))
```

The counter on line 40 ```for i, item in enumerate(self._iterator):``` is checking self.start and self.num_lines, but self._iterator will have been forwarded by the call to next. I expect that above loop will run twice as much as expected.

Comment (zhangguanheng66) [reply torchtext/experimental/datasets/raw/common.py:47]: Add a CI test to cover this case. https://github.com/pytorch/text/pull/1141/files#diff-1154fb0ac2ec7b1fbd994e75fa0e50a3ac102a4f10d0c7bb142047926a83fffeR147

Comment (cpuhrsch) [reply torchtext/experimental/datasets/raw/common.py:47]: Yes, so this holds true then. next is meant to deplete the iterator, but instead the loop runs for the entirety of the dataset.

```
>>> lst = [1, 3, 2, 4, 5]
>>> lst_iter = iter(lst)
>>> next(lst_iter)
1
>>> next(lst_iter)
3
>>> next(lst_iter)
2
>>> next(lst_iter)
4
>>> next(lst_iter)
5
>>> next(lst_iter)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

Comment (zhangguanheng66) [reply torchtext/experimental/datasets/raw/common.py:47]: OK, I tried the similar thing
```
In [8]: from torchtext.experimental.datasets.raw import AG_NEWS
   ...: train_iter, test_iter = AG_NEWS(split=('train', 'test'))
   ...: for i, item in enumerate(train_iter):
   ...:     print("iter ->", item)
   ...:     print("next ->", next(train_iter))
   ...:     if i > 1:
   ...:       break
   ...:
iter -> (3, "Wall St. Bears Claw Back Into the Black (Reuters) Reuters - Short-sellers, Wall Street's dwindling\\band of ultra-cynics, are seeing green again.")
next -> (3, 'Carlyle Looks Toward Commercial Aerospace (Reuters) Reuters - Private investment firm Carlyle Group,\\which has a reputation for making well-timed and occasionally\\controversial plays in the defense industry, has quietly placed\\its bets on another part of the market.')
iter -> (3, "Oil and Economy Cloud Stocks' Outlook (Reuters) Reuters - Soaring crude prices plus worries\\about the economy and the outlook for earnings are expected to\\hang over the stock market next week during the depth of the\\summer doldrums.")
next -> (3, 'Iraq Halts Oil Exports from Main Southern Pipeline (Reuters) Reuters - Authorities have halted oil export\\flows from the main pipeline in southern Iraq after\\intelligence showed a rebel militia could strike\\infrastructure, an oil official said on Saturday.')
iter -> (3, 'Oil prices soar to all-time record, posing new menace to US economy (AFP) AFP - Tearaway world oil prices, toppling records and straining wallets, present a new economic menace barely three months before the US presidential elections.')
next -> (3, 'Stocks End Up, But Near Year Lows (Reuters) Reuters - Stocks ended slightly higher on Friday\\but stayed near lows for the year as oil prices surged past  #36;46\\a barrel, offsetting a positive outlook from computer maker\\Dell Inc. (DELL.O)')
```

The original text file is
```
     1 "3","Wall St. Bears Claw Back Into the Black (Reuters)","Reuters - Short-sellers, Wall Street's dwindling\band of ultra-cynics,        are seeing green again."
     2 "3","Carlyle Looks Toward Commercial Aerospace (Reuters)","Reuters - Private investment firm Carlyle Group,\which has a reputati       on for making well-timed and occasionally\controversial plays in the defense industry, has quietly placed\its bets on another pa       rt of the market."
     3 "3","Oil and Economy Cloud Stocks' Outlook (Reuters)","Reuters - Soaring crude prices plus worries\about the economy and the out       look for earnings are expected to\hang over the stock market next week during the depth of the\summer doldrums."
     4 "3","Iraq Halts Oil Exports from Main Southern Pipeline (Reuters)","Reuters - Authorities have halted oil export\flows from the        main pipeline in southern Iraq after\intelligence showed a rebel militia could strike\infrastructure, an oil official said on Sa       turday."
     5 "3","Oil prices soar to all-time record, posing new menace to US economy (AFP)","AFP - Tearaway world oil prices, toppling recor       ds and straining wallets, present a new economic menace barely three months before the US presidential elections."
     6 "3","Stocks End Up, But Near Year Lows (Reuters)","Reuters - Stocks ended slightly higher on Friday\but stayed near lows for the        year as oil prices surged past  #36;46\a barrel, offsetting a positive outlook from computer maker\Dell Inc. (DELL.O)"
```

File: torchtext/experimental/datasets/raw/common.py [hunk 2]
```diff
@@ -44,6 +44,10 @@ def __iter__(self):
...
+        item = self._iterator.__next__()
```
Comment (cpuhrsch) [inline torchtext/experimental/datasets/raw/common.py:48]: You can just use ```next``` to call this.

**Review by cpuhrsch — APPROVED** [review]

**Pull request merged**